### PR TITLE
Use CompareStringOrdinal in DoFileSearch instead of _wcsicmp

### PR DESCRIPTION
**Pull request by shuffle2**
I was told this is "more correct", but tbh I'm not really sure if it matters for our use case.

### CODE REVIEW

File: Source/Core/Common/FileSearch.cpp [hunk 1]
```diff
@@ -102,12 +104,12 @@ std::vector<std::string> DoFileSearch(const std::vector<std::string>& directorie
     if (recursive)
     {
       // TODO use fs::directory_options::follow_directory_symlink ?
-      for (auto& entry : fs::recursive_directory_iterator(fs::path(directory.c_str())))
+      for (auto& entry : fs::recursive_directory_iterator(fs::path(directory)))
```
Comment (iwubcode) [inline Source/Core/Common/FileSearch.cpp]: This is unrelated?

Comment (shuffle2) [inline Source/Core/Common/FileSearch.cpp]: yea but i'm not in the habbit of creating branches and prs for 1liners
